`test_zeta` is flaky

https://travis-ci.org/chainer/chainer/jobs/587292895#L2445-L2645
``` TestZeta_use_chainerx_false__chainerx_device_None__use_cuda_false__cuda_device_None__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never_param_1_{dtype=float16, shape=(3, 2)}.test_double_backward 

self = <chainer.testing._bundle.TestZeta_use_chainerx_false__chainerx_device_None__use_cuda_false__cuda_device_None__use_cudn..._fast_batch_normalization_false__use_ideep_never_param_1_{dtype=float16, shape=(3, 2)} testMethod=test_double_backward>
backend_config = <BackendConfig use_chainerx=False chainerx_device=None use_cuda=False cuda_device=None use_cudnn='never' cudnn_deterministic=False autotune=False cudnn_fast_batch_normalization=False use_ideep='never'>

    def test_double_backward(self, backend_config):
        """Tests double-backward computation."""
>       self.run_test_double_backward(backend_config)

../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/testing/function_link.py:463: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/testing/function_link.py:313: in run_test_double_backward
    do_check()
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/testing/function_link.py:302: in do_check
    **self.check_double_backward_options)
/opt/python/3.5.6/lib/python3.5/contextlib.py:77: in __exit__
    self.gen.throw(type, value, traceback)
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/testing/function_link.py:36: in raise_if_fail
    cls.fail(message, e)
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/testing/function_link.py:27: in fail
    utils._raise_from(cls, message, exc)
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/utils/__init__.py:106: in _raise_from
    six.raise_from(new_exc.with_traceback(orig_exc.__traceback__), None)
<string>:3: in raise_from
    ???
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/testing/function_link.py:34: in raise_if_fail
    yield
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/testing/function_link.py:302: in do_check
    **self.check_double_backward_options)
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:1049: in check_double_backward
    utils._raise_from(AssertionError, f.getvalue(), e)
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/utils/__init__.py:106: in _raise_from
    six.raise_from(new_exc.with_traceback(orig_exc.__traceback__), None)
<string>:3: in raise_from
    ???
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:1030: in check_double_backward
    detect_nondifferentiable=detect_nondifferentiable)
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:900: in check_backward
    detect_nondifferentiable, is_immutable_params=False
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:464: in run
    self._run()
../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:507: in _run
    self._compare_gradients(gx_numeric, gx_backward, directions)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <chainer.gradient_check._CheckBackward object at 0x7ffa0e36cbe0>, gx_numeric = array(-11.15456357)
gx_backward = -11.14185535922681
directions = [array([[-0.41801665,  0.41556274],
       [-0.34919773, -0.11665035],
       [-0.11101484, -0.00713268]]), array([[ 0.3032942 ,  0.15081198],
       [-0.53963982,  0.2944959 ],
       [-0.10095515,  0.04225671]])]

    def _compare_gradients(self, gx_numeric, gx_backward, directions):
        atol = self.atol
        rtol = self.rtol
        # Compare the gradients
        try:
            testing.assert_allclose(
                gx_numeric, gx_backward, atol=atol, rtol=rtol)
        except AssertionError as e:
            eps = self.eps
            xs = self.xs
            gys = self.gys
            f = six.StringIO()
            f.write('check_backward failed (eps={} atol={} rtol={})\n'.format(
                eps, atol, rtol))
            for i, x in enumerate(xs):
                f.write('inputs[{}]:\n'.format(i))
                f.write('{}\n'.format(x))
            for i, gy in enumerate(gys):
                f.write('grad_outputs[{}]:\n'.format(i))
                f.write('{}\n'.format(gy))
            for i, d in enumerate(directions):
                f.write('directions[{}]:\n'.format(i))
                f.write('{}\n'.format(d))
            f.write('gradients (numeric):  {}\n'.format(gx_numeric))
            f.write('gradients (backward): {}\n'.format(gx_backward))
            f.write('\n')
            f.write('x: numeric gradient, y: backward gradient')
            f.write(str(e))
>           raise AssertionError(f.getvalue())
E           chainer.testing.function_link.FunctionTestError: Parameterized test failed.
E           
E           Base test method: TestZeta_use_chainerx_false__chainerx_device_None__use_cuda_false__cuda_device_None__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never.test_double_backward
E           Test parameters:
E             dtype: <class 'numpy.float16'>
E             shape: (3, 2)
E           
E           
E           (caused by)
E           FunctionTestError: double backward is not implemented correctly
E           
E           (caused by)
E           AssertionError: check_double_backward failed (eps=0.001 atol=0.001 rtol=0.001)
E           input[0]:
E           [[ 1.027 47.97 ]
E            [49.75  36.8  ]
E            [30.23   9.81 ]]
E           grad_output[0]:
E           [[-0.4482 -0.2605]
E            [ 0.987  -0.803 ]
E            [-0.5586 -0.977 ]]
E           grad_grad_input[0]:
E           [[-0.12354 -0.1595 ]
E            [ 0.876    0.7744 ]
E            [-0.731   -0.3467 ]]
E           
E           check_backward failed (eps=0.001 atol=0.001 rtol=0.001)
E           inputs[0]:
E           [[ 1.027 47.97 ]
E            [49.75  36.8  ]
E            [30.23   9.81 ]]
E           inputs[1]:
E           [[-0.4482 -0.2605]
E            [ 0.987  -0.803 ]
E            [-0.5586 -0.977 ]]
E           grad_outputs[0]:
E           [[-0.12354 -0.1595 ]
E            [ 0.876    0.7744 ]
E            [-0.731   -0.3467 ]]
E           directions[0]:
E           [[-0.41801665  0.41556274]
E            [-0.34919773 -0.11665035]
E            [-0.11101484 -0.00713268]]
E           directions[1]:
E           [[ 0.3032942   0.15081198]
E            [-0.53963982  0.2944959 ]
E            [-0.10095515  0.04225671]]
E           gradients (numeric):  -11.154563571250966
E           gradients (backward): -11.14185535922681
E           
E           x: numeric gradient, y: backward gradient
E           Not equal to tolerance rtol=0.001, atol=0.001
E           
E           (mismatch 100.0%)
E            x: array(-11.154564)
E            y: array(-11.141855)
E           
E           assert_allclose failed: 
E             shape: () ()
E             dtype: float64 float64
E             i: (0,)
E             x[i]: -11.154563571250966
E             y[i]: -11.14185535922681
E             relative error[i]: 0.0011405831088654775
E             absolute error[i]: 0.012708212024156396
E             relative tolerance * |y[i]|: 0.01114185535922681
E             absolute tolerance: 0.001
E             total tolerance: 0.012141855359226812
E           x: -11.15456357
E           y: -11.14185536
E           
E           
E           (caused by)
E           AssertionError: check_backward failed (eps=0.001 atol=0.001 rtol=0.001)
E           inputs[0]:
E           [[ 1.027 47.97 ]
E            [49.75  36.8  ]
E            [30.23   9.81 ]]
E           inputs[1]:
E           [[-0.4482 -0.2605]
E            [ 0.987  -0.803 ]
E            [-0.5586 -0.977 ]]
E           grad_outputs[0]:
E           [[-0.12354 -0.1595 ]
E            [ 0.876    0.7744 ]
E            [-0.731   -0.3467 ]]
E           directions[0]:
E           [[-0.41801665  0.41556274]
E            [-0.34919773 -0.11665035]
E            [-0.11101484 -0.00713268]]
E           directions[1]:
E           [[ 0.3032942   0.15081198]
E            [-0.53963982  0.2944959 ]
E            [-0.10095515  0.04225671]]
E           gradients (numeric):  -11.154563571250966
E           gradients (backward): -11.14185535922681
E           
E           x: numeric gradient, y: backward gradient
E           Not equal to tolerance rtol=0.001, atol=0.001
E           
E           (mismatch 100.0%)
E            x: array(-11.154564)
E            y: array(-11.141855)
E           
E           assert_allclose failed: 
E             shape: () ()
E             dtype: float64 float64
E             i: (0,)
E             x[i]: -11.154563571250966
E             y[i]: -11.14185535922681
E             relative error[i]: 0.0011405831088654775
E             absolute error[i]: 0.012708212024156396
E             relative tolerance * |y[i]|: 0.01114185535922681
E             absolute tolerance: 0.001
E             total tolerance: 0.012141855359226812
E           x: -11.15456357
E           y: -11.14185536

../../../virtualenv/python3.5.6/lib/python3.5/site-packages/chainer/gradient_check.py:537: FunctionTestError
```

[Another](https://jenkins.preferred.jp/job/chainer/job/cupy_pr/410/TEST=chainer-py35,label=mn1-p100/testReport/tests.chainer_tests.functions_tests.math_tests.test_zeta/TestZeta_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_1__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never_param_1_%7Bdtype=float16,%20shape=(3,%202)%7D/test_double_backward/)

self = <chainer.gradient_check._CheckBackward object at 0x7fc1229d6fd0>
gx_numeric = array(6.884856481040702), gx_backward = array(6.876974447138121)
directions = [array([[ 0.14789319, -0.39819422],
       [-0.30069263,  0.48494284],
       [-0.13386525,  0.3350874 ]]), array([[ 0.05378703, -0.00259355],
       [ 0.42868435,  0.34321846],
       [ 0.07387417,  0.23206365]])]

    def _compare_gradients(self, gx_numeric, gx_backward, directions):
        atol = self.atol
        rtol = self.rtol
        # Compare the gradients
        try:
            testing.assert_allclose(
                gx_numeric, gx_backward, atol=atol, rtol=rtol)
        except AssertionError as e:
            eps = self.eps
            xs = self.xs
            gys = self.gys
            f = six.StringIO()
            f.write('check_backward failed (eps={} atol={} rtol={})\n'.format(
                eps, atol, rtol))
            for i, x in enumerate(xs):
                f.write('inputs[{}]:\n'.format(i))
                f.write('{}\n'.format(x))
            for i, gy in enumerate(gys):
                f.write('grad_outputs[{}]:\n'.format(i))
                f.write('{}\n'.format(gy))
            for i, d in enumerate(directions):
                f.write('directions[{}]:\n'.format(i))
                f.write('{}\n'.format(d))
            f.write('gradients (numeric):  {}\n'.format(gx_numeric))
            f.write('gradients (backward): {}\n'.format(gx_backward))
            f.write('\n')
            f.write('x: numeric gradient, y: backward gradient')
            f.write(str(e))
>           raise AssertionError(f.getvalue())
E           chainer.testing.function_link.FunctionTestError: Parameterized test failed.
E           
E           Base test method: TestZeta_use_chainerx_false__chainerx_device_None__use_cuda_true__cuda_device_1__use_cudnn_never__cudnn_deterministic_false__autotune_false__cudnn_fast_batch_normalization_false__use_ideep_never.test_double_backward
E           Test parameters:
E             dtype: <class 'numpy.float16'>
E             shape: (3, 2)
E           
E           
E           (caused by)
E           FunctionTestError: double backward is not implemented correctly
E           
E           (caused by)
E           AssertionError: check_double_backward failed (eps=0.001 atol=0.001 rtol=0.001)
E           input[0]:
E           [[  6.16796875   6.5       ]
E            [ 26.546875    31.8125    ]
E            [ 18.546875     1.05957031]]
E           grad_output[0]:
E           [[-0.20947266 -0.55517578]
E            [-0.21911621  0.69287109]
E            [-0.53125     0.23974609]]
E           grad_grad_input[0]:
E           [[ 0.78417969  0.22595215]
E            [ 0.82373047  0.359375  ]
E            [-0.17004395  0.66992188]]
E           
E           check_backward failed (eps=0.001 atol=0.001 rtol=0.001)
E           inputs[0]:
E           [[  6.16796875   6.5       ]
E            [ 26.546875    31.8125    ]
E            [ 18.546875     1.05957031]]
E           inputs[1]:
E           [[-0.20947266 -0.55517578]
E            [-0.21911621  0.69287109]
E            [-0.53125     0.23974609]]
E           grad_outputs[0]:
E           [[ 0.78417969  0.22595215]
E            [ 0.82373047  0.359375  ]
E            [-0.17004395  0.66992188]]
E           directions[0]:
E           [[ 0.14789319 -0.39819422]
E            [-0.30069263  0.48494284]
E            [-0.13386525  0.3350874 ]]
E           directions[1]:
E           [[ 0.05378703 -0.00259355]
E            [ 0.42868435  0.34321846]
E            [ 0.07387417  0.23206365]]
E           gradients (numeric):  6.884856481040702
E           gradients (backward): 6.876974447138121
E           
E           x: numeric gradient, y: backward gradient
E           Not equal to tolerance rtol=0.001, atol=0.001
E           
E           (mismatch 100.0%)
E            x: array(6.884856481040702)
E            y: array(6.876974447138121)
E           
E           assert_allclose failed: 
E             shape: () ()
E             dtype: float64 float64
E             i: (0,)
E             x[i]: 6.884856481040702
E             y[i]: 6.876974447138121
E             relative error[i]: 0.0011461484935225876
E             absolute error[i]: 0.007882033902580687
E             relative tolerance * |y[i]|: 0.006876974447138121
E             absolute tolerance: 0.001
E             total tolerance: 0.00787697444713812
E           x: 6.884856481040702
E           y: 6.876974447138121
E           
E           
E           (caused by)
E           AssertionError: check_backward failed (eps=0.001 atol=0.001 rtol=0.001)
E           inputs[0]:
E           [[  6.16796875   6.5       ]
E            [ 26.546875    31.8125    ]
E            [ 18.546875     1.05957031]]
E           inputs[1]:
E           [[-0.20947266 -0.55517578]
E            [-0.21911621  0.69287109]
E            [-0.53125     0.23974609]]
E           grad_outputs[0]:
E           [[ 0.78417969  0.22595215]
E            [ 0.82373047  0.359375  ]
E            [-0.17004395  0.66992188]]
E           directions[0]:
E           [[ 0.14789319 -0.39819422]
E            [-0.30069263  0.48494284]
E            [-0.13386525  0.3350874 ]]
E           directions[1]:
E           [[ 0.05378703 -0.00259355]
E            [ 0.42868435  0.34321846]
E            [ 0.07387417  0.23206365]]
E           gradients (numeric):  6.884856481040702
E           gradients (backward): 6.876974447138121
E           
E           x: numeric gradient, y: backward gradient
E           Not equal to tolerance rtol=0.001, atol=0.001
E           
E           (mismatch 100.0%)
E            x: array(6.884856481040702)
E            y: array(6.876974447138121)
E           
E           assert_allclose failed: 
E             shape: () ()
E             dtype: float64 float64
E             i: (0,)
E             x[i]: 6.884856481040702
E             y[i]: 6.876974447138121
E             relative error[i]: 0.0011461484935225876
E             absolute error[i]: 0.007882033902580687
E             relative tolerance * |y[i]|: 0.006876974447138121
E             absolute tolerance: 0.001
E             total tolerance: 0.00787697444713812
E           x: 6.884856481040702
E           y: 6.876974447138121

atol       = 0.001
d          = array([[ 0.05378703, -0.00259355],
       [ 0.42868435,  0.34321846],
       [ 0.07387417,  0.23206365]])
directions = [array([[ 0.14789319, -0.39819422],
       [-0.30069263,  0.48494284],
       [-0.13386525,  0.3350874 ]]), array([[ 0.05378703, -0.00259355],
       [ 0.42868435,  0.34321846],
       [ 0.07387417,  0.23206365]])]
eps        = 0.001
f          = <_io.StringIO object at 0x7fc122a1bca8>
gx_backward = array(6.876974447138121)
gx_numeric = array(6.884856481040702)
gy         = array([[ 0.78417969,  0.22595215],
       [ 0.82373047,  0.359375  ],
       [-0.17004395,  0.66992188]], dtype=float16)
gys        = (array([[ 0.78417969,  0.22595215],
       [ 0.82373047,  0.359375  ],
       [-0.17004395,  0.66992188]], dtype=float16),)
i          = 1
rtol       = 0.001
self       = <chainer.gradient_check._CheckBackward object at 0x7fc1229d6fd0>
x          = array([[-0.20947266, -0.55517578],
       [-0.21911621,  0.69287109],
       [-0.53125   ,  0.23974609]], dtype=float16)
xs         = (array([[  6.16796875,   6.5       ],
       [ 26.546875  ,  31.8125    ],
       [ 18.546875  ,   1.05957031]], dtype...ray([[-0.20947266, -0.55517578],
       [-0.21911621,  0.69287109],
       [-0.53125   ,  0.23974609]], dtype=float16))

--- a/tests/chainer_tests/functions_tests/math_tests/test_zeta.py
+++ b/tests/chainer_tests/functions_tests/math_tests/test_zeta.py
@@ -9,7 +9,9 @@ from chainer import testing
 
 @testing.parameterize(*testing.product({
     'dtype': [numpy.float16, numpy.float32, numpy.float64],
-    'shape': [(), (3, 2)]
+    'shape': [(), (3, 2)],
+    'x_range': [(1.1, 2), (2, 50)],
+    'q_range': [(1.1, 2), (2, 50)],
 }))
 @testing.inject_backend_tests(
     None,
@@ -32,16 +34,23 @@ from chainer import testing
 class TestZeta(testing.FunctionTestCase):
 
     def setUp(self):
-        self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-3}
-        self.check_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
-        self.check_double_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
+        if self.dtype == numpy.float16:
+            self.check_forward_options = {'atol': 1e-2, 'rtol': 1e-2}
+            self.check_backward_options = {'atol': 1e-2, 'rtol': 1e-2}
+            self.check_double_backward_options = {'atol': 1e-2, 'rtol': 1e-2}
+        else:
+            self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-3}
+            self.check_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
+            self.check_double_backward_options = {'atol': 1e-3, 'rtol': 1e-3}
 
-        self._x = numpy.random.uniform(low=2, high=50, size=self.shape).\
-            astype(self.dtype)
+        low, high = self.x_range
+        self._x = numpy.random.uniform(
+            low=low, high=high, size=self.shape).astype(self.dtype)
 
     def generate_inputs(self):
-        q = numpy.random.uniform(low=1, high=50, size=self.shape).\
-            astype(self.dtype)
+        low, high = self.q_range
+        q = numpy.random.uniform(
+            low=low, high=high, size=self.shape).astype(self.dtype)
         return q,
 
     def forward_expected(self, inputs):

